File: src/Domestica.Budget.Application/Transactions/AddInternalTransaction/AddInternalTransactionCommand.cs
using CommonAbstractions.DB.Messaging;
using Domestica.Budget.Domain.Transactions;

namespace Domestica.Budget.Application.Transactions.AddInternalTransaction
{
    public sealed record AddInternalTransactionCommand(
        string FromAccountId,
        string ToAccountId,
        decimal TransactionAmount,
        DateTime TransactionDateTime) : ICommand<Transaction[]>
    {
    }
}

File: src/client-app/src/models/requests/addInternalTransactionCommand.ts
export interface AddInternalTransactionCommand {
    fromAccountId: string
    toAccountId: string
    transactionAmount: number
    transactionDateTime: Date
}

export class AddInternalTransactionCommand implements AddInternalTransactionCommand {
    constructor(values?: AddInternalTransactionCommand) {
        this.fromAccountId = values?.fromAccountId || ''
        this.toAccountId = values?.toAccountId || ''
        this.transactionAmount = values?.transactionAmount || 0
        this.transactionDateTime = values?.transactionDateTime || new Date()
    }
}